wagtailcache/cache.py

```diff
@@ -337,15 +337,19 @@ def process_response(
                 # (of the chopped request, not the real one).
                 cr = _chop_querystring(request)
                 uri = unquote(cr.build_absolute_uri())
-                keyring = self._wagcache.get("keyring", {})
-                # Get current cache keys belonging to this URI.
-                # This should be a list of keys.
-                uri_keys: List[str] = keyring.get(uri, [])
-                # Append the key to this list if not already present and save.
-                if cache_key not in uri_keys:
-                    uri_keys.append(cache_key)
-                    keyring[uri] = uri_keys
-                    self._wagcache.set("keyring", keyring)
+
+                if wagtailcache_settings.WAGTAIL_CACHE_KEYRING:
+                    keyring = self._wagcache.get("keyring", {})
+                    # Get current cache keys belonging to this URI.
+                    # This should be a list of keys.
+                    uri_keys: List[str] = keyring.get(uri, [])
+                    limit = wagtailcache_settings.WAGTAIL_CACHE_KEYRING_LIMIT
+                    # Append the key to this list if not already present and save.
+                    if cache_key not in uri_keys and len(keyring) < limit:
```

Comment:
Let's find a way to support having an "unlimited" keyring i.e. `WAGTAIL_CACHE_KEYRING_LIMIT = None`. I'd also run this check before checking `cache_key not in uri_keys` to avoid the dictionary lookup operation if we are over the limit.